File: interfaceTest/ssros_ams/src/gesture_subscriber.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "test_interfaces/msg/head_touch_gesture.hpp"

void topic_callback(const test_interfaces::msg::HeadTouchGesture::SharedPtr msg)
{
  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "I heard: gesture=%d, value=%d, direction=%d", msg->gesture, msg->value, msg->direction);
}

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = rclcpp::Node::make_shared("gesture_subscriber");
  auto subscription = node->create_subscription<test_interfaces::msg::HeadTouchGesture>(
    "head_touch_gesture", 10, topic_callback);
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: interfaceTest/ams/ssros_ams_py/ssros_ams_py/gesture_subscriber.py
import rclpy
from rclpy.node import Node
from test_interfaces.msg import HeadTouchGesture


class GestureSubscriber(Node):
    def __init__(self):
        super().__init__("gesture_subscriber_py")
        self.subscription = self.create_subscription(
            HeadTouchGesture, "head_touch_gesture", self.listener_callback, 10
        )
        self.subscription  # prevent unused variable warning

    def listener_callback(self, msg):
        self.get_logger().info(
            'I heard: gesture="%d", value="%d", direction="%d"'
            % (msg.gesture, msg.value, msg.direction)
        )


def main(args=None):
    rclpy.init(args=args)
    gesture_subscriber = GestureSubscriber()
    rclpy.spin(gesture_subscriber)
    gesture_subscriber.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
